### PR TITLE
Raise exception on MPS write error

### PR DESCRIPTION
**Pull request by pet-mit**
Issue described here: https://github.com/rte-france/or-tools/issues/44

### CODE REVIEW

**Comment by flomnes** [comment]
I suggest defining your own exception, derived from `std::runtime_error`

This way, the caller can catch both `std::runtime_error`, but also this more specific new exception.

```cc
class MPSWriteError : public std::runtime_error {
// ...
};
```

**Comment by flomnes** [comment]
Even though we only use Sirius and XPRESS, it would be nice for consistency that other solvers also `throw` in case of error.

**Comment by pet-mit** [comment]
> ```c++
> class MPSWriteError : public std::runtime_error {
> // ...
> };
> ```

I agree, but are we sure we want to diverge from google's code for these interfaces that we do not master?

**Comment by pet-mit** [comment]
> Even though we only use Sirius and XPRESS, it would be nice for consistency that other solvers also `throw` in case of error.

I agree, but are we sure we want to diverge from google's code for these interfaces that we do not master?

File: ortools/linear_solver/sirius_interface.cc [hunk 1]
```diff
@@ -1406,7 +1407,7 @@ namespace operations_research {
 		VLOG(1) << "Writing Sirius MPS \"" << filename << "\".";
 		const int status = SRSwritempsprob(mLp->problem_mps, filename.c_str());
 		if (status) {
-			LOG(WARNING) << "Failed to write MPS.";
+                  MPSWriteError("Failed to write MPS.");
```
Comment (flomnes) [inline ortools/linear_solver/sirius_interface.cc:1410]:
```suggestion
                  throw MPSWriteError("Failed to write MPS.");
```

File: ortools/linear_solver/MPSWriteError.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+// Initial version of this code was provided by RTE
+
+#ifndef ORTOOLS_MPSWRITEERROR_H
+#define ORTOOLS_MPSWRITEERROR_H
+#include <stdexcept>
+namespace operations_research {
+class MPSWriteError : public std::runtime_error {
+ public:
+  MPSWriteError(std::string message) : std::runtime_error(message) {}
```
Comment (flomnes) [inline ortools/linear_solver/MPSWriteError.h:22]: Log message here ?

```suggestion
  MPSWriteError(const std::string& message) : std::runtime_error(message) {}
```

Comment (pet-mit) [reply ortools/linear_solver/MPSWriteError.h:22]: since we're throwing an exception, I think we shouldn't log it.

Comment (flomnes) [reply ortools/linear_solver/MPSWriteError.h]: If the user doesn't write something like
```cpp
try {
solver->Write(...);
} catch (const std::runtime_error& ex) {
// ...
}
```
The behavior will be an error message printed in `std::cerr` such as

> terminate called after throwing an instance of 'MPSWriteError'
> Abandon (core dumped)

If you think this is enough, we can keep it like this. Keep in mind that in some cases, `std::cerr` may not be captured, only logs.

Comment (pet-mit) [reply ortools/linear_solver/MPSWriteError.h]: I do think that it's up to the user in such a case to catch & log the exception. But if you strongly feel against it, we can add the log

Comment (flomnes) [reply ortools/linear_solver/MPSWriteError.h]: OK, we can keep it like this. But then we have to document this (`Write` function may throw XXX)

**Review by flomnes — APPROVED** [review]

**Pull request merged**